be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -253,9 +254,13 @@ object DbActor extends AskPatternConstants {
             //FIXME: use some sort of compare-and-swap to prevent concurrency issues with LaoData modification if needed
             if(bt != null){
               val laoJson = new String(bt, StandardCharsets.UTF_8)
-              batch.put(laoDataKey.getBytes, LaoData.buildFromJson(laoJson).updateWith(message).toJsonString.getBytes)
+              val laoDataNew: LaoData = LaoData.buildFromJson(laoJson).updateWith(message)
+              batch.put(laoDataKey.getBytes, laoDataNew.toJsonString.getBytes)
+              laoDataCache.put(laoDataKey, laoDataNew)
             } else if (bt == null) {
-              batch.put(laoDataKey.getBytes, LaoData.emptyLaoData.updateWith(message).toJsonString.getBytes)
+              val laoDataNew: LaoData = LaoData.emptyLaoData.updateWith(message)
+              batch.put(laoDataKey.getBytes, laoDataNew.toJsonString.getBytes)
```

Comment:
Could `laoDataKey.getBytes` be null and hence has to be treated as an edge case. ?

Reply:
And it would be great to have some modularization for future caching of other objects , don't you think so ?

Reply:
It could be useful, however, for now, the object we need the most (by far) is LaoData, whether it's for "access rights" (is a person allowed to send a message) or for server-side broadcasts. The other structures will be used less, however, caching them in the future could be an option.

Reply:
In the current implementation, I don't see it being possible, however for the code to be more robust for the future, I'll add extra checks